src/components/BurnModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import { Mesh } from "../contracts/Mesh";
+import { Button, Modal } from "../components";
+
+type Props = {
+  isModalOpen: boolean;
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
this is good, buy fyi you can just do `(isOpen: boolean) => void`

Reply:
no need to change this btw

Reply:
Nice - just figured it would be safer to strong type the states considering that's how I'm controlling opening and closing the modal